### PR TITLE
resolves #98 cannot set image alternative for accessibility

### LINKED ISSUE #98: Cannot set image alternative for accessibility
In cards, the attribute alt of the image is always set with the title of the card. Sometimes images show more than a title and we cannot describe it without changing the title.

Example :
image
image

This issue is linked to another issue in Tock and Tock Studio in order to give an extra parameter description of a file.

### PR DESCRIPTION
**Pull request by correi-f**
resolves #98 cannot set image alternative for accessibility

### CODE REVIEW

File: src/components/Image/Image.tsx [hunk 1]
```diff
@@ -45,18 +46,19 @@ export const CardContainer: StyledComponent<
...
+  { title, url, alternative }: ImageProps,
   ref,
 ) {
   return (
     <CardOuter ref={ref}>
       <CardContainer>
         {url && (
           <a target="_blank" href={url} rel="noreferrer">
-            <CardImage src={url} alt={title} />
+            <CardImage src={url} alt={alternative ? alternative : title} />
```
Comment (Fabilin) [inline src/components/Image/Image.tsx:61]:
```suggestion
            <CardImage src={url} alt={alternative ?? title} />
```

File: src/useTock.ts [hunk 2]
```diff
@@ -72,15 +72,17 @@ function mapCard(card: any): Card {
     title: card.title,
     subTitle: card.subTitle,
     imageUrl: card.file ? card.file.url : null,
+    imageAlternative: card.file ? card.file.description : null,
```
Comment (Fabilin) [inline src/useTock.ts:75]:
```suggestion
    imageUrl: card.file?.url,
    imageAlternative: card?.file.description,
```

File: src/components/Card/Card.tsx [hunk 3]
```diff
@@ -143,7 +144,7 @@ const Card = React.forwardRef<HTMLLIElement, CardProps>(function cardRender(
       }
     >
       <CardContainer>
-        {imageUrl && <CardImage src={imageUrl} alt={title} />}
+        {imageUrl && <CardImage src={imageUrl} alt={imageAlternative ? imageAlternative : title} />}
```
Comment (Fabilin) [inline src/components/Card/Card.tsx:147]:
```suggestion
        {imageUrl && <CardImage src={imageUrl} alt={imageAlternative ?? title} />}
```

**Review by Fabilin — COMMENTED** [review]
Looks good to me, was just thinking some bits of code could be simplified.

File: src/components/Image/Image.tsx [hunk 4]
```diff
@@ -45,18 +46,19 @@ export const CardContainer: StyledComponent<
...
+  { title, url, alternative }: ImageProps,
   ref,
 ) {
   return (
     <CardOuter ref={ref}>
       <CardContainer>
         {url && (
           <a target="_blank" href={url} rel="noreferrer">
-            <CardImage src={url} alt={title} />
+            <CardImage src={url} alt={alternative ?? title} />
```
Comment (elebescond) [inline src/components/Image/Image.tsx:61]: Possibilité de déplacer la logique du fallback dans la function mapImage

Comment (correi-f) [reply src/components/Image/Image.tsx]: Done

File: src/components/Card/Card.tsx [hunk 5]
```diff
@@ -143,7 +144,7 @@ const Card = React.forwardRef<HTMLLIElement, CardProps>(function cardRender(
       }
     >
       <CardContainer>
-        {imageUrl && <CardImage src={imageUrl} alt={title} />}
+        {imageUrl && <CardImage src={imageUrl} alt={imageAlternative ?? title} />}
```
[the remark this answers is not in the document]
Comment (correi-f) [reply src/components/Card/Card.tsx]: Done

**Review by Fabilin — APPROVED** [review]

**Pull request merged**